connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala

```diff
@@ -139,7 +139,18 @@ object SparkConnectServerUtils {
       .map(clientTestJar => Seq("--jars", clientTestJar.getCanonicalPath))
       .getOrElse(Seq.empty)
 
-    writerV2Configs ++ hiveTestConfigs ++ udfTestConfigs
```

Comment:
@LuciferYang would you mind help checking this part? I am not familiar with this

Reply:
ok

Reply:
run the following commands:
```
build/sbt clean
build/sbt "connect-client-jvm/test" -Phive
```

there is 1 test failed:

```
[info] - call_function *** FAILED *** (150 milliseconds)
[info]   org.apache.spark.SparkException: [CANNOT_LOAD_FUNCTION_CLASS] Cannot load class test.org.apache.spark.sql.MyDoubleSum when registering the function `spark_catalog`.`default`.`custom_sum`, please make sure it is on the classpath.
[info]   at org.apache.spark.sql.connect.client.GrpcExceptionConverter$.toSparkThrowable(GrpcExceptionConverter.scala:53)
[info]   at org.apache.spark.sql.connect.client.GrpcExceptionConverter$.convert(GrpcExceptionConverter.scala:30)
[info]   at org.apache.spark.sql.connect.client.GrpcExceptionConverter$$anon$1.hasNext(GrpcExceptionConverter.scala:38)
[info]   at org.apache.spark.sql.connect.client.SparkResult.org$apache$spark$sql$connect$client$SparkResult$$processResponses(SparkResult.scala:80)
[info]   at org.apache.spark.sql.connect.client.SparkResult.length(SparkResult.scala:133)
[info]   at org.apache.spark.sql.connect.client.SparkResult.toArray(SparkResult.scala:150)
[info]   at org.apache.spark.sql.Dataset.$anonfun$collect$1(Dataset.scala:2813)
[info]   at org.apache.spark.sql.Dataset.withResult(Dataset.scala:3252)
[info]   at org.apache.spark.sql.Dataset.collect(Dataset.scala:2812)
[info]   at org.apache.spark.sql.ClientE2ETestSuite.$anonfun$new$139(ClientE2ETestSuite.scala:1175)
[info]   at org.apache.spark.sql.connect.client.util.RemoteSparkSession.$anonfun$test$1(RemoteSparkSession.scala:246)
```

@beliefer we should add `(LocalProject("sql") / Test / Keys.`package`).value` to 

https://github.com/apache/spark/blob/228b5dbfd7688a8efa7135d9ec7b00b71e41a38a/project/SparkBuild.scala#L875-L878

then the sql test jar will build&package before testing.

Reply:
@LuciferYang Thank you for you reminder. I will add it.

Reply:
maven test `ClientE2ETestSuite` and `ReplE2ESuite` is ok, but there are another 68 maven tests failed of `connect-jvm-client` module, will tracking with a new ticket. @zhengruifeng

Reply:
@LuciferYang  Thank you!

Reply:
hmm... please wait a moment, let me check this again.

Reply:
re-check maven on master and with-this-pr:

**Before**

```
build/mvn clean -Phadoop-3 -Phadoop-cloud -Pmesos -Pyarn -Pkinesis-asl -Phive-thriftserver -Pspark-ganglia-lgpl -Pkubernetes -Phive
build/mvn clean install -DskipTests -Phive
build/mvn clean test -pl connector/connect/client/jvm
```

```
Run completed in 2 minutes, 26 seconds.
Total number of tests run: 1099
Suites: completed 24, aborted 0
Tests: succeeded 1099, failed 0, canceled 0, ignored 1, pending 0
All tests passed.
```

**After**

```
gh pr checkout 41932
build/mvn clean -Phadoop-3 -Phadoop-cloud -Pmesos -Pyarn -Pkinesis-asl -Phive-thriftserver -Pspark-ganglia-lgpl -Pkubernetes -Phive
build/mvn clean install -DskipTests -Phive
build/mvn clean test -pl connector/connect/client/jvm
```
Run completed in 1 minute, 29 seconds.
Total number of tests run: 1101
Suites: completed 24, aborted 0
Tests: succeeded 1033, failed 68, canceled 0, ignored 1, pending 0
*** 68 TESTS FAILED ***
```

A bit magical, but seems this pr has caused other cases to maven test fail. Could you double check this ? @beliefer

Reply:
@LuciferYang I tested with
```
build/mvn clean install -DskipTests -Phadoop-3 -Phadoop-cloud -Pmesos -Pyarn -Pkinesis-asl -Phive-thriftserver -Pspark-ganglia-lgpl -Pkubernetes -Phive
build/mvn test -pl connector/connect/client/jvm
```
I reproduce this issue too. It seems is unrelated to this PR.
```
Run completed in 2 minutes, 10 seconds.
Total number of tests run: 1101
Suites: completed 24, aborted 0
Tests: succeeded 1033, failed 68, canceled 0, ignored 1, pending 0
*** 68 TESTS FAILED ***
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:02 min
[INFO] Finished at: 2023-07-18T20:20:38+08:00
```

Reply:
> @LuciferYang I tested with
> 
> ```
> build/mvn clean install -DskipTests -Phadoop-3 -Phadoop-cloud -Pmesos -Pyarn -Pkinesis-asl -Phive-thriftserver -Pspark-ganglia-lgpl -Pkubernetes -Phive
> build/mvn test -pl connector/connect/client/jvm
> ```
> 
> I reproduce this issue too. It seems is unrelated to this PR.
> 
> ```
> Run completed in 2 minutes, 10 seconds.
> Total number of tests run: 1101
> Suites: completed 24, aborted 0
> Tests: succeeded 1033, failed 68, canceled 0, ignored 1, pending 0
> *** 68 TESTS FAILED ***
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  03:02 min
> [INFO] Finished at: 2023-07-18T20:20:38+08:00
> ```

master branch?

Reply:
I'm checking now.

Reply:
To clarify, in my local testing:
- master branch: All tests passed.
- with this pr: 68 TESTS FAILED

Reply:
I think this is truly unrelated to this pr and I think the way the “–jars”  is being used in the code is incorrect now.

When submitting the args as

```
--jars spark-catalyst-xx.jar
--jars spark-connect-client-jvm-xx.jar
--jars spark-sql-xx.jar
```

the final effective arg will be `--jars spark-sql-xx.jar`, if we enable debugging logs, we will found that only the “Added JAR” logs related to “spark-sql_2.12-3.5.0-SNAPSHOT-tests.jar” and “spark-connect_2.12-3.5.0-SNAPSHOT.jar” are present.

```
23/07/19 14:00:34 INFO SparkContext: Added JAR file:///Users/yangjie01/SourceCode/git/spark-mine-12/sql/core/target/spark-sql_2.12-3.5.0-SNAPSHOT-tests.jar at spark://localhost:56841/jars/spark-sql_2.12-3.5.0-SNAPSHOT-tests.jar with timestamp 1689746434318
23/07/19 14:00:34 INFO SparkContext: Added JAR file:/Users/yangjie01/SourceCode/git/spark-mine-12/connector/connect/server/target/spark-connect_2.12-3.5.0-SNAPSHOT.jar at spark://localhost:56841/jars/spark-connect_2.12-3.5.0-SNAPSHOT.jar with timestamp 1689746434318
```

and the configuration item “spark.jars” also only includes these two jars.

```
Array((spark.app.name,org.apache.spark.sql.connect.SimpleSparkConnectService), (spark.jars,file:///Users/yangjie01/SourceCode/git/spark-mine-12/sql/core/target/spark-sql_2.12-3.5.0-SNAPSHOT-tests.jar,file:/Users/yangjie01/SourceCode/git/spark-mine-12/connector/connect/server/target/spark-connect_2.12-3.5.0-SNAPSHOT.jar), ...
```

We should correct the usage of “–jars” to “--jars spark-catalyst-xx.jar,spark-connect-client-jvm-xx.jar,spark-sql-xx.jar”, then the maven test should pass.

I think we can merge this pr first and then fix this issue separately. But, @beliefer if you prefer, you can also address this issue in this one :)

Reply:
@LuciferYang Thank you for the investigation. I will take it.

Reply:
Both maven and sbt ok now, thanks @beliefer

Reply:
Thank you for the double check. @LuciferYang